data_diff/__init__.py

```diff
@@ -1,9 +1,86 @@
-from typing import Tuple
+from typing import Tuple, Iterator
 
 from .database import connect_to_uri
-from .diff_tables import TableSegment, TableDiffer
+from .diff_tables import (
+    TableSegment,
+    TableDiffer,
+    DEFAULT_BISECTION_THRESHOLD,
+    DEFAULT_BISECTION_FACTOR,
+    DbKey,
+    DbTime,
+    parse_table_name,
+)
 
 
-def create_source(db_uri: str, table_name: str, key_column: str, extra_columns: Tuple[str, ...] = ()):
-    db = connect_to_uri(db_uri)
-    return TableSegment(db, (table_name,), key_column, tuple(extra_columns))
+class TableRef:
```

Comment:
Remind me the reasoning why we introduce a `TableRef` on top of `TableSegment`? 👂🏻

Reply:
TableSegment is complicated to construct, and it is used more as an internal structure. 99% of the users don't care about start_key, and if they give a start_time they'll only want to give it once, instead of duplicate it for each TableSegment.

Users can still use it, but we should offer a simple and easy API for the common use case.